### DISCUSSION TITLE
Meaning of “Blocked by CNAME or IP”

### DISCUSSION THREAD

**Discussion by laaubert** [AdguardTeam/AdGuardHome#5002] [Q&A]
Hi,

I'm curious to understand if it is expected that few requests being blocked by AdGuard Home are also sent to the upstream DNS server which in my case will block them as well:

AdGuard logs:

<img width="884" alt="image" src="https://user-images.githubusercontent.com/50697061/194051882-e83a5730-607d-4d96-92bb-36d5dd8b6dad.png">

Upstream server logs (nextDNS):

![image](https://user-images.githubusercontent.com/50697061/194053490-164ea9f2-eee3-4747-b5bf-9c28de2382b5.png)

Thanks !!

**Comment by ainar-g** [comment]
“Blocked by CNAME or IP” means that it's not the request that is in a blocklist, but rather a response.  That is, AdGuard Home has performed a request for a domain that isn't in any of the blocklists, but the response contains IP addresses or canonical domain names that **are** blocked.

**Reply by laaubert** [reply]
Thanks for the quick reply !!!

nextDNS is saying it used the  "Lightswitch05 - Ads & Tracking" list to block this domain. 

<img width="469" alt="image" src="https://user-images.githubusercontent.com/50697061/194055532-9bae5774-32f1-452b-90ee-52f15678c883.png">

I have the same list installed on AdGuard and when I do a check, it can't find it:

<img width="660" alt="image" src="https://user-images.githubusercontent.com/50697061/194055037-ebe0e32a-fc1c-4c67-93a3-c5669fbb3e7d.png">

Furthermore if I search this domain in the list itself, it is not listed which makes me doubt about the filter info logged by nextDNS...

Another observation is why did AdGuard specify a filter list in its log if the domain didn't match any ? Will such reply be cached like any other regular DNS response coming from the upstream DNS server ?

Thanks again for your support
